src/runner.ts

```diff
@@ -4,17 +4,36 @@ import { DocumentInventory } from './documentInventory';
 import { lintNode } from './linter';
 import { getDiagnosticSeverity } from './rulesets/severity';
 import { FunctionCollection, IGivenNode, IRule, IRuleResult, IRunRule, RunRuleCollection } from './types';
+import { RulesetExceptionCollection } from './types/ruleset';
 import { hasIntersectingElement } from './utils/';
+import { IExceptionLocation, pivotExceptions } from './utils/pivotExceptions';
 
 export const isRuleEnabled = (rule: IRule) => rule.severity !== void 0 && getDiagnosticSeverity(rule.severity) !== -1;
 
 export const runRules = (
   documentInventory: DocumentInventory,
   rules: RunRuleCollection,
   functions: FunctionCollection,
+  exceptions: RulesetExceptionCollection,
 ): IRuleResult[] => {
   const results: IRuleResult[] = [];
 
+  // TODO:
+  // Ruleset
+  // - To be investigated: How to cope with STDIN as a source and exception relative locations?
```

Comment:
@P0lip As I'm not sure how to build a ruleset that would be able to reference a document passed through STDIN, I'm wondering if issuing a warning wouldn't be the best short-term approach.

When the ruleset contains exceptions
And the document is passed as STDIN
A warning would be issued notifying that exceptions handling isn't supported in that mode and that the run will ignore them.

Thoughts?

Reply:
If you STDIN a file, it may well still have $ref's, and those $ref's would still have exceptions working just the same right? It'd just be the initial "entry file" being passed in as STDIN, and that could have things ignored with `{ENTRY FILE OR ROOT IDENTIFIER}#/components` and not mentioning a filename? Or do we need some sort of wildcard for saying "this is the first/main/root/entry file?

Reply:
@philsturgeon I'm not even sure how a $ref containing a relative file  location would be resolved in that context and I can't find any test in the codebase about that use case. Does it consider the current working directory as the root to be used?

@P0lip Any idea?

Reply:
IIRC, the document path would be equal to the output of `process.cwd()`.
I wouldn't really bother too much though, as it's kind of expected you provide a dereferenced document as STDIN.

Reply:
yeah CWD.

Reply:
🎉 It took me some time (as Windows and tty don't play well together) but I managed to find a way to honor exceptions in a `stdin` context.

@P0lip @philsturgeon Could you please take a look at the proposal?

Reply:
As a side note, that more or less compelled me to find a sustainable way to develop in a xplat way from a Windows computer. I'm going to open a series of tiny pull requests to make contributing from Windows a bit easier.